<a href="https://colab.research.google.com/github/Ashvin-Vargez/Telecom-Churn-Analysis-EDA-Project-/blob/main/eda_Telecom_Churn_Analysis_Capstone_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## <b> Orange S.A., formerly France Télécom S.A., is a French multinational telecommunications corporation. The Orange Telecom's Churn Dataset, consists of cleaned customer activity data (features), along with a churn label specifying whether a customer canceled the subscription.

## <b> Explore and analyze the data to discover key factors responsible for customer churn and come up with ways/recommendations to ensure customer retention. </b>

In [1]:
#importing necessary libraries for data manipulation and visualization 
import numpy as np 
import pandas as pd 
import datetime 
import seaborn as sns 
import matplotlib.pyplot as plt 
%matplotlib inline

In [2]:
# importing drive functionality to read files from google drive  
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [3]:
#reading csv file and creating pandas dataframe using pd.read_csv
df_tel = pd.read_csv('/content/drive/MyDrive/Almabetter/EDA Capstone Project _churn analysis/Telecom Churn.csv')

In [4]:
#checking the created dataframe
df_tel.head()

State  Account length  Area code International plan Voice mail plan  \
0    KS             128        415                 No             Yes   
1    OH             107        415                 No             Yes   
2    NJ             137        415                 No              No   
3    OH              84        408                Yes              No   
4    OK              75        415                Yes              No   

   Number vmail messages  Total day minutes  Total day calls  \
0                     25              265.1              110   
1                     26              161.6              123   
2                      0              243.4              114   
3                      0              299.4               71   
4                      0              166.7              113   

   Total day charge  Total eve minutes  Total eve calls  Total eve charge  \
0             45.07              197.4               99             16.78   
1             27.47              195.5              103             16.62   
2             41.38              121.2              110             10.30   
3             50.90               61.9               88              5.26   
4             28.34              148.3              122             12.61   

   Total night minutes  Total night calls  Total night charge  \
0                244.7                 91               11.01   
1                254.4                103               11.45   
2                162.6                104                7.32   
3                196.9                 89                8.86   
4                186.9                121                8.41   

   Total intl minutes  Total intl calls  Total intl charge  \
0                10.0                 3               2.70   
1                13.7                 3               3.70   
2                12.2                 5               3.29   
3                 6.6                 7               1.78   
4                10.1                 3               2.73   

   Customer service calls  Churn  
0                       1  False  
1                       1  False  
2                       0  False  
3                       2  False  
4                       3  False

In [5]:
#checking the tail of the dataframe 
df_tel.tail()

State  Account length  Area code International plan Voice mail plan  \
3328    AZ             192        415                 No             Yes   
3329    WV              68        415                 No              No   
3330    RI              28        510                 No              No   
3331    CT             184        510                Yes              No   
3332    TN              74        415                 No             Yes   

      Number vmail messages  Total day minutes  Total day calls  \
3328                     36              156.2               77   
3329                      0              231.1               57   
3330                      0              180.8              109   
3331                      0              213.8              105   
3332                     25              234.4              113   

      Total day charge  Total eve minutes  Total eve calls  Total eve charge  \
3328             26.55              215.5              126             18.32   
3329             39.29              153.4               55             13.04   
3330             30.74              288.8               58             24.55   
3331             36.35              159.6               84             13.57   
3332             39.85              265.9               82             22.60   

      Total night minutes  Total night calls  Total night charge  \
3328                279.1                 83               12.56   
3329                191.3                123                8.61   
3330                191.9                 91                8.64   
3331                139.2                137                6.26   
3332                241.4                 77               10.86   

      Total intl minutes  Total intl calls  Total intl charge  \
3328                 9.9                 6               2.67   
3329                 9.6                 4               2.59   
3330                14.1                 6               3.81   
3331                 5.0                10               1.35   
3332                13.7                 4               3.70   

      Customer service calls  Churn  
3328                       2  False  
3329                       3  False  
3330                       2  False  
3331                       2  False  
3332                       0  False

In [6]:
#checking the shape of dataframe 
df_tel.shape

(3333, 20)

#Here we can see that the total number of rows is 3333 and that there are 20 columns associated with each row.


In [7]:
#getting an overview of the dataframe 
df_tel.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3333 entries, 0 to 3332
Data columns (total 20 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   State                   3333 non-null   object 
 1   Account length          3333 non-null   int64  
 2   Area code               3333 non-null   int64  
 3   International plan      3333 non-null   object 
 4   Voice mail plan         3333 non-null   object 
 5   Number vmail messages   3333 non-null   int64  
 6   Total day minutes       3333 non-null   float64
 7   Total day calls         3333 non-null   int64  
 8   Total day charge        3333 non-null   float64
 9   Total eve minutes       3333 non-null   float64
 10  Total eve calls         3333 non-null   int64  
 11  Total eve charge        3333 non-null   float64
 12  Total night minutes     3333 non-null   float64
 13  Total night calls       3333 non-null   int64  
 14  Total night charge      3333 non-null   

In [8]:
#checking the dataframe for null values 
df_tel.isnull().sum()

State                     0
Account length            0
Area code                 0
International plan        0
Voice mail plan           0
Number vmail messages     0
Total day minutes         0
Total day calls           0
Total day charge          0
Total eve minutes         0
Total eve calls           0
Total eve charge          0
Total night minutes       0
Total night calls         0
Total night charge        0
Total intl minutes        0
Total intl calls          0
Total intl charge         0
Customer service calls    0
Churn                     0
dtype: int64

#From the above operation we found out that there are no null values in our data set, therefore we dont have to do any additional operations to handle missing values.

In [9]:
#checking characteristics of the dataframe 
df_tel_stat = df_tel.describe().transpose()
df_tel_stat

count        mean        std     min     25%     50%  \
Account length          3333.0  101.064806  39.822106    1.00   74.00  101.00   
Area code               3333.0  437.182418  42.371290  408.00  408.00  415.00   
Number vmail messages   3333.0    8.099010  13.688365    0.00    0.00    0.00   
Total day minutes       3333.0  179.775098  54.467389    0.00  143.70  179.40   
Total day calls         3333.0  100.435644  20.069084    0.00   87.00  101.00   
Total day charge        3333.0   30.562307   9.259435    0.00   24.43   30.50   
Total eve minutes       3333.0  200.980348  50.713844    0.00  166.60  201.40   
Total eve calls         3333.0  100.114311  19.922625    0.00   87.00  100.00   
Total eve charge        3333.0   17.083540   4.310668    0.00   14.16   17.12   
Total night minutes     3333.0  200.872037  50.573847   23.20  167.00  201.20   
Total night calls       3333.0  100.107711  19.568609   33.00   87.00  100.00   
Total night charge      3333.0    9.039325   2.275873    1.04    7.52    9.05   
Total intl minutes      3333.0   10.237294   2.791840    0.00    8.50   10.30   
Total intl calls        3333.0    4.479448   2.461214    0.00    3.00    4.00   
Total intl charge       3333.0    2.764581   0.753773    0.00    2.30    2.78   
Customer service calls  3333.0    1.562856   1.315491    0.00    1.00    1.00   

                           75%     max  
Account length          127.00  243.00  
Area code               510.00  510.00  
Number vmail messages    20.00   51.00  
Total day minutes       216.40  350.80  
Total day calls         114.00  165.00  
Total day charge         36.79   59.64  
Total eve minutes       235.30  363.70  
Total eve calls         114.00  170.00  
Total eve charge         20.00   30.91  
Total night minutes     235.30  395.00  
Total night calls       113.00  175.00  
Total night charge       10.59   17.77  
Total intl minutes       12.10   20.00  
Total intl calls          6.00   20.00  
Total intl charge         3.27    5.40  
Customer service calls    2.00    9.00

In [10]:
#creating a dataframe with data exculsively of churned customers and loyal customers
df_churn = df_tel[df_tel['Churn'] == True ] 
df_loyal = df_tel[df_tel['Churn']== False]

In [11]:
#checking for insights from churn dataframe 
churn_stat = df_churn.describe().transpose()
churn_stat

count        mean        std     min      25%     50%  \
Account length          483.0  102.664596  39.467820    1.00   76.000  103.00   
Area code               483.0  437.817805  42.792270  408.00  408.000  415.00   
Number vmail messages   483.0    5.115942  11.860138    0.00    0.000    0.00   
Total day minutes       483.0  206.914079  68.997792    0.00  153.250  217.60   
Total day calls         483.0  101.335404  21.582307    0.00   87.500  103.00   
Total day charge        483.0   35.175921  11.729710    0.00   26.055   36.99   
Total eve minutes       483.0  212.410145  51.728910   70.90  177.100  211.30   
Total eve calls         483.0  100.561077  19.724711   48.00   87.000  101.00   
Total eve charge        483.0   18.054969   4.396762    6.03   15.055   17.96   
Total night minutes     483.0  205.231677  47.132825   47.40  171.250  204.80   
Total night calls       483.0  100.399586  19.950659   49.00   85.000  100.00   
Total night charge      483.0    9.235528   2.121081    2.13    7.705    9.22   
Total intl minutes      483.0   10.700000   2.793190    2.00    8.800   10.60   
Total intl calls        483.0    4.163561   2.551575    1.00    2.000    4.00   
Total intl charge       483.0    2.889545   0.754152    0.54    2.380    2.86   
Customer service calls  483.0    2.229814   1.853275    0.00    1.000    2.00   

                            75%     max  
Account length          127.000  225.00  
Area code               510.000  510.00  
Number vmail messages     0.000   48.00  
Total day minutes       265.950  350.80  
Total day calls         116.500  165.00  
Total day charge         45.210   59.64  
Total eve minutes       249.450  363.70  
Total eve calls         114.000  168.00  
Total eve charge         21.205   30.91  
Total night minutes     239.850  354.90  
Total night calls       115.000  158.00  
Total night charge       10.795   15.97  
Total intl minutes       12.800   20.00  
Total intl calls          5.000   20.00  
Total intl charge         3.460    5.40  
Customer service calls    4.000    9.00

In [12]:
#checking for insights from loyal dataframe
loyal_stat = df_loyal.describe().transpose()
loyal_stat

count        mean        std     min       25%  \
Account length          2850.0  100.793684  39.882350    1.00   73.0000   
Area code               2850.0  437.074737  42.306156  408.00  408.0000   
Number vmail messages   2850.0    8.604561  13.913125    0.00    0.0000   
Total day minutes       2850.0  175.175754  50.181655    0.00  142.8250   
Total day calls         2850.0  100.283158  19.801157    0.00   87.0000   
Total day charge        2850.0   29.780421   8.530835    0.00   24.2825   
Total eve minutes       2850.0  199.043298  50.292175    0.00  164.5000   
Total eve calls         2850.0  100.038596  19.958414    0.00   87.0000   
Total eve charge        2850.0   16.918909   4.274863    0.00   13.9800   
Total night minutes     2850.0  200.133193  51.105032   23.20  165.9000   
Total night calls       2850.0  100.058246  19.506246   33.00   87.0000   
Total night charge      2850.0    9.006074   2.299768    1.04    7.4700   
Total intl minutes      2850.0   10.158877   2.784489    0.00    8.4000   
Total intl calls        2850.0    4.532982   2.441984    0.00    3.0000   
Total intl charge       2850.0    2.743404   0.751784    0.00    2.2700   
Customer service calls  2850.0    1.449825   1.163883    0.00    1.0000   

                           50%     75%     max  
Account length          100.00  127.00  243.00  
Area code               415.00  510.00  510.00  
Number vmail messages     0.00   22.00   51.00  
Total day minutes       177.20  210.30  315.60  
Total day calls         100.00  114.00  163.00  
Total day charge         30.12   35.75   53.65  
Total eve minutes       199.60  233.20  361.80  
Total eve calls         100.00  114.00  170.00  
Total eve charge         16.97   19.82   30.75  
Total night minutes     200.25  234.90  395.00  
Total night calls       100.00  113.00  175.00  
Total night charge        9.01   10.57   17.77  
Total intl minutes       10.20   12.00   18.90  
Total intl calls          4.00    6.00   19.00  
Total intl charge         2.75    3.24    5.10  
Customer service calls    1.00    2.00    8.00

In [13]:
df_churn_percent = (churn_stat/df_tel_stat)*100
df_churn_percent

count        mean         std         min  \
Account length          14.491449  101.582935   99.110330  100.000000   
Area code               14.491449  100.145337  100.993548  100.000000   
Number vmail messages   14.491449   63.167499   86.643934         NaN   
Total day minutes       14.491449  115.096074  126.677251         NaN   
Total day calls         14.491449  100.895857  107.540070         NaN   
Total day charge        14.491449  115.095765  126.678468         NaN   
Total eve minutes       14.491449  105.687022  102.001554         inf   
Total eve calls         14.491449  100.446255   99.006583         inf   
Total eve charge        14.491449  105.686342  101.997246         inf   
Total night minutes     14.491449  102.170357   93.196044  204.310345   
Total night calls       14.491449  100.291561  101.952360  148.484848   
Total night charge      14.491449  102.170549   93.198555  204.807692   
Total intl minutes      14.491449  104.519810  100.048361         inf   
Total intl calls        14.491449   92.948085  103.671402         inf   
Total intl charge       14.491449  104.520144  100.050349         inf   
Customer service calls  14.491449  142.675541  140.880847         NaN   

                               25%         50%         75%         max  
Account length          102.702703  101.980198  100.000000   92.592593  
Area code               100.000000  100.000000  100.000000  100.000000  
Number vmail messages          NaN         NaN    0.000000   94.117647  
Total day minutes       106.645790  121.293200  122.897412  100.000000  
Total day calls         100.574713  101.980198  102.192982  100.000000  
Total day charge        106.651658  121.278689  122.886654  100.000000  
Total eve minutes       106.302521  104.915591  106.013600  100.000000  
Total eve calls         100.000000  101.000000  100.000000   98.823529  
Total eve charge        106.320621  104.906542  106.025000  100.000000  
Total night minutes     102.544910  101.789264  101.933702   89.848101  
Total night calls        97.701149  100.000000  101.769912   90.285714  
Total night charge      102.460106  101.878453  101.935788   89.870568  
Total intl minutes      103.529412  102.912621  105.785124  100.000000  
Total intl calls         66.666667  100.000000   83.333333  100.000000  
Total intl charge       103.478261  102.877698  105.810398  100.000000  
Customer service calls  100.000000  200.000000  200.000000  100.000000

Note that the average number of customer service calls placed by the churned users are 142.67 % of the average customer service calls and the median number of calls are 200% of the number of customer service calls placed by the total users. 

#From the above table, we can see that 14.49 % of the total users have churned and that the users placing a significantly higher number of custmer service calls than the median are at risk of churn.



#This also indicates that the services were not satisfactory for the churned customers.

# Even though the average and median number of international calls made by the churned customers were less than the genral population, the average and median number of minutes of international calls were higher. But the charges incurred by the churned population is not significantly higher than the overall user base.

This indicates that the churned customers made international calls of longer duration than the general population, which may indicate that the users have family or relatives overseas and that the calls may be personal calls and not business calls. 

#Thus the reason for their churn may be motivated by quality or connectivity issues on international calls rather than incurring a higher cost for the same.  

In [14]:
# comparing statistics form the churned dataset to that of the loyal dataset 
churn_loyal_percent = (churn_stat/loyal_stat)*100
churn_loyal_percent

count        mean         std         min  \
Account length          16.947368  101.856180   98.960619  100.000000   
Area code               16.947368  100.170009  101.149038  100.000000   
Number vmail messages   16.947368   59.456163   85.244241         NaN   
Total day minutes       16.947368  118.117989  137.496046         NaN   
Total day calls         16.947368  101.049275  108.995183         NaN   
Total day charge        16.947368  118.117609  137.497786         NaN   
Total eve minutes       16.947368  106.715547  102.856776         inf   
Total eve calls         16.947368  100.522279   98.829050         inf   
Total eve charge        16.947368  106.714737  102.851543         inf   
Total night minutes     16.947368  102.547545   92.227365  204.310345   
Total night calls       16.947368  100.341142  102.278313  148.484848   
Total night charge      16.947368  102.547772   92.230217  204.807692   
Total intl minutes      16.947368  105.326601  100.312462         inf   
Total intl calls        16.947368   91.850368  104.487804         inf   
Total intl charge       16.947368  105.326996  100.314955         inf   
Customer service calls  16.947368  153.798861  159.232020         NaN   

                               25%         50%         75%         max  
Account length          104.109589  103.000000  100.000000   92.592593  
Area code               100.000000  100.000000  100.000000  100.000000  
Number vmail messages          NaN         NaN    0.000000   94.117647  
Total day minutes       107.299142  122.799097  126.462197  111.153359  
Total day calls         100.574713  103.000000  102.192982  101.226994  
Total day charge        107.299496  122.808765  126.461538  111.164958  
Total eve minutes       107.659574  105.861723  106.968268  100.525152  
Total eve calls         100.000000  101.000000  100.000000   98.823529  
Total eve charge        107.689557  105.833824  106.987891  100.520325  
Total night minutes     103.224834  102.272160  102.107280   89.848101  
Total night calls        97.701149  100.000000  101.769912   90.285714  
Total night charge      103.145917  102.330744  102.128666   89.870568  
Total intl minutes      104.761905  103.921569  106.666667  105.820106  
Total intl calls         66.666667  100.000000   83.333333  105.263158  
Total intl charge       104.845815  104.000000  106.790123  105.882353  
Customer service calls  100.000000  200.000000  200.000000  112.500000

#Analysis based on States 

In [15]:
#creating a dataframe with states and their respective churn percentages 
churn_state = pd.DataFrame(df_churn['State'].value_counts())
total_state = pd.DataFrame(df_tel['State'].value_counts())
state_percent = pd.merge(churn_state, total_state, how = 'inner', left_index=True, right_index=True )
state_percent['churn_percentage'] = (state_percent['State_x']/state_percent['State_y'])*100
state_percent = state_percent.sort_values('churn_percentage', ascending = False)
state_percent.rename( columns = {'State_x':'Churn_no', 'State_y':'total_no'}, inplace = True)
print(state_percent.head())
print(state_percent.tail())

    Churn_no  total_no  churn_percentage
NJ        18        68         26.470588
CA         9        34         26.470588
TX        18        72         25.000000
MD        17        70         24.285714
SC        14        60         23.333333
    Churn_no  total_no  churn_percentage
IA         3        44          6.818182
VA         5        77          6.493506
AZ         4        64          6.250000
AK         3        52          5.769231
HI         3        53          5.660377


From the above table, we can see that the states with five highest churn rates are NJ, CA, TX, MD, SC.
The states with five lowest churn rates are HI, AK, AZ, VA and IA.

In [16]:
# checking quick statistics 
state_percent.describe().transpose()

count       mean        std        min        25%  \
Churn_no           51.0   9.470588   4.001764   3.000000   6.500000   
total_no           51.0  65.352941  11.801396  34.000000  59.500000   
churn_percentage   51.0  14.492688   5.757934   5.660377   9.555691   

                        50%        75%         max  
Churn_no           9.000000  12.500000   18.000000  
total_no          65.000000  72.500000  106.000000  
churn_percentage  13.636364  18.321859   26.470588

In [17]:
#create dfs for high churn states and low churn states 

In [18]:
# checking number of service calls made by the churned customers 
median_calls = pd.merge(df_churn.groupby('State')['Customer service calls'].median(), df_loyal.groupby('State')['Customer service calls'].median(), how = 'inner', left_index = True, right_index = True)
median_calls.head()
# try to get a frequency plot with number of service calls made 
# try to compare with service calls made by customers of smae state who didnt churn 


Customer service calls_x  Customer service calls_y
State                                                    
AK                          4.0                       1.0
AL                          2.5                       1.0
AR                          4.0                       1.5
AZ                          2.0                       1.0
CA                          0.0                       1.0

In [19]:
median_calls.rename(columns = { 'Customer service calls_x':'calls by churned cust', 'Customer service calls_y':'calls by loyal cust'}, inplace = True)
median_calls.head()

calls by churned cust  calls by loyal cust
State                                            
AK                       4.0                  1.0
AL                       2.5                  1.0
AR                       4.0                  1.5
AZ                       2.0                  1.0
CA                       0.0                  1.0

In [25]:
median_calls['difference'] = median_calls['calls by churned cust']- median_calls['calls by loyal cust']
median_calls[median_calls['difference']<=.5]

calls by churned cust  calls by loyal cust  difference
State                                                        
CA                       0.0                  1.0        -1.0
CT                       1.0                  1.0         0.0
DC                       1.0                  1.0         0.0
DE                       1.0                  2.0        -1.0
GA                       2.0                  2.0         0.0
IL                       1.0                  1.0         0.0
KS                       1.0                  1.0         0.0
MA                       1.0                  1.0         0.0
ME                       2.0                  2.0         0.0
MS                       1.5                  1.0         0.5
ND                       1.0                  1.0         0.0
NE                       1.0                  1.0         0.0
NV                       1.5                  1.0         0.5
OH                       0.5                  1.0        -0.5
OK                       1.0                  2.0        -1.0
PA                       1.5                  1.0         0.5
SD                       1.5                  1.0         0.5
TX                       1.0                  1.0         0.0
VA                       2.0                  1.5         0.5
WA                       1.5                  1.0         0.5
WI                       1.0                  1.0         0.0
WY                       1.0                  1.0         0.0

In [27]:
mean_calls = pd.merge(df_churn.groupby('State')['Customer service calls'].mean(), df_loyal.groupby('State')['Customer service calls'].mean(), how = 'inner', left_index = True, right_index = True)
mean_calls.head()

Customer service calls_x  Customer service calls_y
State                                                    
AK                     3.333333                  1.448980
AL                     3.000000                  1.402778
AR                     3.818182                  1.522727
AZ                     2.250000                  1.550000
CA                     1.777778                  1.360000

In [30]:
mean_calls['difference'] = mean_calls['Customer service calls_x']- mean_calls['Customer service calls_y']
mean_calls[mean_calls['difference']<=1]

Customer service calls_x  Customer service calls_y  difference
State                                                                
AZ                     2.250000                  1.550000    0.700000
CA                     1.777778                  1.360000    0.417778
CO                     2.555556                  1.666667    0.888889
CT                     1.416667                  1.532258   -0.115591
DC                     1.800000                  1.306122    0.493878
DE                     1.444444                  1.653846   -0.209402
FL                     2.125000                  1.490909    0.634091
GA                     2.250000                  1.869565    0.380435
IL                     1.200000                  1.132075    0.067925
KS                     1.692308                  1.315789    0.376518
LA                     2.250000                  1.510638    0.739362
MA                     1.363636                  1.185185    0.178451
MD                     2.235294                  1.528302    0.706992
ME                     2.000000                  1.673469    0.326531
MS                     2.285714                  1.529412    0.756303
NC                     2.272727                  1.421053    0.851675
ND                     1.500000                  1.446429    0.053571
NE                     1.200000                  1.482143   -0.282143
NJ                     2.166667                  1.500000    0.666667
NM                     2.500000                  1.500000    1.000000
NV                     2.142857                  1.442308    0.700549
OH                     1.000000                  1.411765   -0.411765
OK                     1.666667                  1.807692   -0.141026
OR                     2.363636                  1.626866    0.736771
PA                     1.875000                  1.135135    0.739865
RI                     2.000000                  1.355932    0.644068
SC                     2.071429                  1.434783    0.636646
SD                     1.500000                  1.384615    0.115385
TX                     1.666667                  1.592593    0.074074
VA                     1.600000                  1.597222    0.002778
WA                     2.071429                  1.384615    0.686813
WI                     1.857143                  1.380282    0.476861
WY                     1.888889                  1.382353    0.506536

In [73]:
# checking customers who have churned without making a customer service call 
df_no_call_churn = df_churn[df_churn['Customer service calls']== 0 ]
df_state_no_call_no = df_no_call_churn['State'].value_counts()
no_call_churn_percent = pd.merge(df_state_no_call_no, churn_state, how = 'inner', left_index = True, right_index = True )
#churn_state is the dataframe containing states and the  respective number of churned customers 
no_call_churn_percent.head()
# try to group no call churners by state and see if there are any relations, eg better plans available

State_x  State_y
TX        5       18
CA        5        9
OH        5       10
DE        4        9
MA        4       11

In [76]:
no_call_churn_percent.rename(columns = {'State_x':' no_call_churns', 'State_y':'total_churns'}, inplace = True)
no_call_churn_percent.head()

no_call_churns  total_churns
TX                5            18
CA                5             9
OH                5            10
DE                4             9
MA                4            11

In [80]:
no_call_churn_percent['percent']= (no_call_churn_percent[['no_call_churns']]/no_call_churn_percent[['total_churns']])*100)
no_call_churn_percent

SyntaxError: ignored

In [ ]:
#binning into categories to classify and make comparisons 
# see which state, plan type, total day minutes, total day calls , charges etc 